src/Functions/registerFunctions.cpp

```diff
@@ -1,145 +1,60 @@
-    registerFunctionsMakeDate(factory);
-    registerFunctionsMap(factory);
-    registerFunctionsBitmap(factory);
-    registerFunctionsBinaryRepr(factory);
-    registerFunctionsCoding(factory);
-    registerFunctionsCodingUUID(factory);
-    registerFunctionChar(factory);
-    registerFunctionsComparison(factory);
-    registerFunctionsConditional(factory);
-    registerFunctionsConversion(factory);
-    registerFunctionCastOrDefault(factory);
-    registerFunctionsDateTime(factory);
-    registerFunctionsEmbeddedDictionaries(factory);
-    registerFunctionsExternalDictionaries(factory);
-    registerFunctionsExternalModels(factory);
-    registerFunctionsFormatting(factory);
-    registerFunctionsHashing(factory);
-    registerFunctionsHigherOrder(factory);
-    registerFunctionsLogical(factory);
-    registerFunctionsMiscellaneous(factory);
-    registerFunctionsRandom(factory);
-    registerFunctionsReinterpret(factory);
-    registerFunctionsRound(factory);
-    registerFunctionsString(factory);
-    registerFunctionsStringArray(factory);
-    registerFunctionsStringSearch(factory);
-    registerFunctionsStringRegexp(factory);
-    registerFunctionsStringSimilarity(factory);
-    registerFunctionsStringTokenExtractor(factory);
-    registerFunctionsURL(factory);
-    registerFunctionsVisitParam(factory);
-    registerFunctionsMath(factory);
-    registerFunctionsGeo(factory);
-    registerFunctionsNull(factory);
-    registerFunctionsJSON(factory);
-    registerFunctionsSQLJSON(factory);
-    registerFunctionToJSONString(factory);
-    registerFunctionsIntrospection(factory);
-    registerFunctionsConsistentHashing(factory);
-    registerFunctionsUnixTimestamp64(factory);
-    registerFunctionBitHammingDistance(factory);
-    registerFunctionTupleHammingDistance(factory);
-    registerFunctionsStringHash(factory);
-    registerFunctionValidateNestedArraySizes(factory);
-    registerFunctionsSnowflake(factory);
-    registerFunctionsTimeWindow(factory);
-    registerFunctionToBool(factory);
-    registerFunctionMinSampleSize(factory);
-
-    //meilisearch
-    registerFunctionMeiliMatch(factory);
-
-#if USE_SSL
-    registerFunctionEncrypt(factory);
-    registerFunctionDecrypt(factory);
-    registerFunctionAESEncryptMysql(factory);
-    registerFunctionAESDecryptMysql(factory);
-    registerFunctionShowCertificate(factory);
-#endif
-    registerFunctionTid(factory);
-    registerFunctionLogTrace(factory);
-    registerFunctionHashID(factory);
+    // These two functions are not build with clickhouse_function,
+    // need to be registered separately
+    earlyRegisterFunctionReplicate(factory);
+    earlyRegisterFunctionsLogical(factory);
+
+    for (const auto &s : symbols) {
+        /* match all functions start with registerFunction */
+        constexpr auto match = "register";
+        /* together with prefix of void DB:: */
+        constexpr auto prefix = "_ZN2DB";
+
+        if (0 != strncmp(s.name, prefix, strlen(prefix)))
```

Comment:
Use `string_view` and `starts_with` instead. Use `string_view` inside this function to avoid `C`-like code.

Reply:
I just tested changing `Symbol::name` from `const char *` to `std::string_view`. There is no noticeable performance difference when running `SYSTEM RELOAD SYMBOLS`

Reply:
The total duration is 84-92 milliseconds before, and increased to 193-202 ms after change to `std::string_view`. Code changed as below. Please let me know shall I proceed to use `std::string_view`
```
#if 1
        auto view = std::string_view(s.name);
        if (!view.starts_with(prefix))
            continue;

        view.remove_prefix(strlen(prefix));
        /* get length of function name */
        char *name_start;
        uint64_t len = std::strtoul(view.data(), &name_start, 10);

#else
        if (0 != strncmp(s.name, prefix, strlen(prefix)))
            continue;

        /* get length of function name */
        char *name_start;
        uint64_t len = std::strtoul(s.name + strlen(prefix), &name_start, 10);
#endif
```